### ISSUE TITLE
Plex Sync and Excessive Logs

### ISSUE THREAD

**Issue by Brando47** [tidusjar/Ombi#3044]
**Describe the bug**
Everything has been working fairly well, however I have been trying to manually trigger a plex recently added sync via a script that triggers plexrecentlyadded in the API. I can do it via the Swagger page, by manually triggering in the settings UI, and the cron job. I have been trying to use the curl suggested on the swagger page: curl -X POST "http://10.1.1.250:47000/api/v1/Job/plexrecentlyadded" -H "accept: application/json" -H "ApiKey: -MyApiKey-"

This may be user error on my behalf however, I'm fairly new to it. In trying to correct my issue, I brought up the logs and noticed I have 700MB over the last week! I have switch the log level to Information (it seemed to be Debug by default) and that seems to have helped, however I am getting a recurring error very frequently that is filling the files.

**To Reproduce**
The logging error seems to be when I've been messing trying to access the API, after some amount of hours it does seem to eventually die off.

**Expected behavior**
Running the sync in ways that has worked has updated the Ombi library successfully and shows up in the logs, my curl command is doing nothing.

**Logs (Logs directory where Ombi is located)**
The following log I got after triggering the sync by a working method, the last [Error] "Something bad happened, ErrorMiddleware caught this" then repeats indefinitely:

> 2019-07-01 09:45:43.983 +09:30 [Debug] Batch acquisition of 0 triggers
> 2019-07-01 09:45:51.381 +09:30 [Debug] Batch acquisition of 1 triggers
> 2019-07-01 09:45:51.384 +09:30 [Debug] Batch acquisition of 0 triggers
> 2019-07-01 09:45:51.385 +09:30 [Debug] Calling Execute on job Plex.IPlexContentSync
> 2019-07-01 09:45:51.403 +09:30 [Information] Starting Plex Content Cacher
> 2019-07-01 09:45:51.408 +09:30 [Information] Starting to cache the content on server "removed"
> 2019-07-01 09:45:53.170 +09:30 [Information] Starting EP Cacher
> 2019-07-01 09:45:53.172 +09:30 [Debug] Batch acquisition of 1 triggers
> 2019-07-01 09:45:53.173 +09:30 [Debug] Batch acquisition of 0 triggers
> 2019-07-01 09:45:53.173 +09:30 [Debug] Calling Execute on job Plex.IPlexEpisodeSync
> 2019-07-01 09:45:53.174 +09:30 [Error] Job Plex.IPlexContentSync threw an unhandled Exception: 
> System.ArgumentNullException: Value cannot be null.
> Parameter name: source
>    at System.Linq.Enumerable.Count[TSource](IEnumerable`1 source)
>    at Ombi.Schedule.Jobs.Plex.PlexContentSync.Execute(IJobExecutionContext context) in C:\projects\requestplex\src\Ombi.Schedule\Jobs\Plex\PlexContentSync.cs:line 119
>    at Quartz.Core.JobRunShell.Run(CancellationToken cancellationToken)
> 2019-07-01 09:45:53.207 +09:30 [Error] Job Plex.IPlexContentSync threw an exception.
> Quartz.SchedulerException: Job threw an unhandled exception. ---> System.ArgumentNullException: Value cannot be null.
> Parameter name: source
>    at System.Linq.Enumerable.Count[TSource](IEnumerable`1 source)
>    at Ombi.Schedule.Jobs.Plex.PlexContentSync.Execute(IJobExecutionContext context) in C:\projects\requestplex\src\Ombi.Schedule\Jobs\Plex\PlexContentSync.cs:line 119
>    at Quartz.Core.JobRunShell.Run(CancellationToken cancellationToken)
>    --- End of inner exception stack trace --- [See nested exception: System.ArgumentNullException: Value cannot be null.
> Parameter name: source
>    at System.Linq.Enumerable.Count[TSource](IEnumerable`1 source)
>    at Ombi.Schedule.Jobs.Plex.PlexContentSync.Execute(IJobExecutionContext context) in C:\projects\requestplex\src\Ombi.Schedule\Jobs\Plex\PlexContentSync.cs:line 119
>    at Quartz.Core.JobRunShell.Run(CancellationToken cancellationToken)]
> 2019-07-01 09:45:53.208 +09:30 [Debug] Trigger instruction : DeleteTrigger
> 2019-07-01 09:45:53.209 +09:30 [Debug] Deleting trigger
> 2019-07-01 09:45:53.251 +09:30 [Information] Total Epsiodes found for TV Shows = 1691
> 2019-07-01 09:45:53.422 +09:30 [Information] Processed 150 more episodes. Total Remaining 1541
> 2019-07-01 09:45:53.540 +09:30 [Information] Processed 150 more episodes. Total Remaining 1391
> 2019-07-01 09:45:53.638 +09:30 [Information] Processed 150 more episodes. Total Remaining 1241
> 2019-07-01 09:45:53.740 +09:30 [Information] Processed 150 more episodes. Total Remaining 1091
> 2019-07-01 09:45:53.843 +09:30 [Information] Processed 150 more episodes. Total Remaining 941
> 2019-07-01 09:45:53.944 +09:30 [Information] Processed 150 more episodes. Total Remaining 791
> 2019-07-01 09:45:54.047 +09:30 [Information] Processed 150 more episodes. Total Remaining 641
> 2019-07-01 09:45:54.158 +09:30 [Information] Processed 150 more episodes. Total Remaining 491
> 2019-07-01 09:45:54.267 +09:30 [Information] Processed 150 more episodes. Total Remaining 341
> 2019-07-01 09:45:54.373 +09:30 [Information] Processed 150 more episodes. Total Remaining 191
> 2019-07-01 09:45:54.409 +09:30 [Information] Processed 150 more episodes. Total Remaining 41
> 2019-07-01 09:45:54.409 +09:30 [Information] We have finished caching the episodes.
> 2019-07-01 09:45:54.410 +09:30 [Debug] Trigger instruction : DeleteTrigger
> 2019-07-01 09:45:54.410 +09:30 [Debug] Batch acquisition of 1 triggers
> 2019-07-01 09:45:54.410 +09:30 [Debug] Deleting trigger
> 2019-07-01 09:45:54.412 +09:30 [Debug] Batch acquisition of 0 triggers
> 2019-07-01 09:45:54.412 +09:30 [Debug] Calling Execute on job Plex.IPlexAvailabilityChecker
> 2019-07-01 09:45:54.743 +09:30 [Debug] Trigger instruction : DeleteTrigger
> 2019-07-01 09:45:54.743 +09:30 [Debug] Deleting trigger
> 2019-07-01 09:45:55.172 +09:30 [Error] Something bad happened, ErrorMiddleware caught this
> System.ArgumentNullException: Value cannot be null.
> Parameter name: source
>    at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source)
>    at Ombi.Controllers.ImagesController.GetBackgroundImage() in C:\projects\requestplex\src\Ombi\Controllers\ImagesController.cs:line 213
>    at lambda_method(Closure , Object )
>    at Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
>    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeActionMethodAsync()
>    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeNextActionFilterAsync()
>    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)
>    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
>    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync()
>    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextResourceFilter()
>    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
>    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
>    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()
>    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
>    at Microsoft.AspNetCore.Builder.RouterMiddleware.Invoke(HttpContext httpContext)
>    at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke(HttpContext context)
>    at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIIndexMiddleware.Invoke(HttpContext httpContext)
>    at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext)
>    at Ombi.ApiKeyMiddlewear.Invoke(HttpContext context) in C:\projects\requestplex\src\Ombi\ApiKeyMiddlewear.cs:line 51
>    at Ombi.ErrorHandlingMiddleware.Invoke(HttpContext context) in C:\projects\requestplex\src\Ombi\ErrorHandlingMiddlewear.cs:line 24

**Desktop (please complete the following information):**
 - Ubuntu Server 18.04

**Ombi Version (please complete the following information):**
 - Ombi 3.0.4639
- Plex Media Server 1.16.1.1291
- Sonarr V3 and Radarr

**Comment by Brando47** [comment]
OK so I've had another tinker tonight, that last [Error] "Something bad happened, ErrorMiddleware caught this" that I thought was causing my logs to fill up hasn't been an issue since I left it for a few hours and it stopped repeating and I can now no longer replicate it. I've managed to successfully run my plex recently added POST request with:
`curl http://10.1.1.250:47000/api/v1/Job/plexrecentlyadded -H "ApiKey: -MyApiKey" -d '{ "accept": "application/json" }'`
however I am still getting the error each time starting with 

> [Error] Job Plex.IPlexContentSync threw an unhandled Exception: System.ArgumentNullException: Value cannot be null.

 ... which I'm not sure if I need to worry about, it is working and catching new content from Plex so I don't really have functionality issues now.

However now that my logs have cleared up somewhat, I've noticed that every time a Radarr sync is done I am getting "[Error] TMDBId is not > 0 for movie "2 Fast 2 Furious"" which repeats for my whole movie collection. Also, if I have debug level logging set (as was default for me after installing), it also prints the whole output from the Radarr api GET /movie for my whole collection, adding about 2MB a pop to my log file every hour when it runs, which is what I think was contributing heavily to my large logs. I've left my log level on Information now to mitigate the issue.

I'm not so sure what the Error is about, I can't quite figure out if it's an ombi or radarr issue. I've tried a few things to clear it like refreshing metadata and rescanning library in radarr and ombi, as well as deleting my OmbiExternal.db. I've noticed in that file that the Radarr cache after syncing only contains movies I've added after installing Radarr/Ombi, and for these the TMDBId Error is absent from the logs (hopefully ombi won't re-request anything, haven't seen anything yet). Checking the Radarr API myself I see that the tmdbid is listed for each file without any problems so I'm not sure why ombi isn't picking it up (perhaps a case-sensitivity issue?).

Either way everything is functional for me now and running smoothly, this app is the icing on the cake for me and my Plex users and I'll be sure to drop a donation as well as get all my users to buy the app :)

**Comment by goldenpipes** [comment]
possibly related to this:
https://www.reddit.com/r/Ombi/comments/cad4gt/error_tmdbid_is_not_0_for_movie/

**Comment by Brando47** [comment]
I had a good chat with that gentleman and we pooled our findings, he eventually figured out that he had exactly the amount of TVDBID errors as he had unmonitored movies in Radarr, hardly seems a coincidence. I'm not sure if he managed to figure out how to stop it from there.

Personally, I have switched from a bare metal install to a docker setup (running a slightly newer version, but the same db files) and my errors have completely disappeared, log files are very clean even after heavy usage of ombi/radarr/sonarr/plex etc.

**Comment by bowser8302** [comment]
So I stumbled on a potential fix for this. 

Make a copy of your "OmbiExternal.db" file (/etc/Ombi/Ombi.External.db in Ubuntu) and delete the original. Now monitor the log file with "tail -f /etc/Ombi/Logs/log-XXXX.txt".

On Ombi's web interface, go into Settings >> Media Server >> Plex. Scroll down and click "Load Libraries" to refresh your library list. Assuming the refresh doesn't throw an error, go through and pick and choose the libraries you want synced. Now click the "Manually Run Full Sync" button. The log you're monitoring should start populating. A few errors were thrown during my runs, but Ombi was successfully synced with Plex and my Movies and TV shows were now appearing in Ombi regardless of the aforementioned errors. 

If I had to guess, this procedure needs to be run on Ombi anytime ANY changes are made to the available libraries in Plex (adding, removing, renaming, etc.) as it seems that Ombi doesn't automatically query/refresh the list of available Plex libraries. So anytime a call to the Plex server is made with an invalid library name/number, Plex throws and error and Ombi, in turn, errors out.
